modules/firefox/hm.nix

```diff
@@ -1,36 +1,66 @@
-# Consider also updating the LibreWolf module when updating this module,
-# as they are very similar.
-
-{ config, lib, ... }:
+{ config, lib, ...}:
 
 let
-  profileSettings = {
-    settings = {
-      "font.name.monospace.x-western" = config.stylix.fonts.monospace.name;
-      "font.name.sans-serif.x-western" = config.stylix.fonts.sansSerif.name;
-      "font.name.serif.x-western" = config.stylix.fonts.serif.name;
-    };
-  };
-  makeProfileSettingsPair = profileName:
-    lib.nameValuePair profileName profileSettings;
-  derivatives = [
-    {path = "firefox"; name = "Firefox";}
-    {path = "librewolf"; name = "LibreWolf";}
+  targets = [
+    { path = "firefox"; name = "Firefox"; }
+    { path = "librewolf"; name = "LibreWolf"; }
   ];
+  eachConfig = mkCfg: targets: lib.mkMerge (map mkCfg targets);
+  eachTarget = mkCfg: lib.mkIf config.stylix.enable (eachConfig (target: let
```

Comment:
This `eachConfig` and `eachTarget` pattern is rather complicated. Do you think it is somehow possible to simplify the code? Note that I only skimmed through the code without a deep inspection.

Reply:
Inlining the functions would work, but that would result in considerable code duplication and IMO less readable code. If this pattern, which btw is commonly used in flake frameworks as `eachSystem`, is too complicated I'll be happy to do that though.

Deduplicating the inline function doesn't work as described here https://github.com/danth/stylix/blob/8b35e355551655bfb6e74d1e615a8a995dab4756/modules/firefox/hm.nix#L30-L31

Basically, `config = lib.mkMerge ...` doesn't work, because `lib.mkMerge` might set `config.stylix.targets.${target}.profileNames`, which then results in an infinite recursion error. The solution I implemented is to only set subfields of `config`, but the linked discourse thread lists some more involved solutions.

Reply:
The `eachConfig` method was rather included for the sake of completeness after adding `eachTarget`. It can be removed at the cost of only small amounts of code duplication.

Reply:
LGTM.